EKrut-Server/src/ekrut/server/EKrutServer.java

```diff
@@ -0,0 +1,199 @@
+import ekrut.net.InventoryItemResponse;
+import ekrut.net.OrderRequest;
+import ekrut.net.OrderResponse;
+import ekrut.net.ShipmentRequest;
+import ekrut.net.ShipmentResponse;
+import ekrut.net.TicketRequest;
+import ekrut.net.TicketResponse;
+import ekrut.net.UserRequest;
+import ekrut.net.UserRequestType;
+import ekrut.net.UserResponse;
+import ekrut.server.db.DBController;
+import ekrut.server.managers.ServerInventoryManager;
+import ekrut.server.managers.ServerOrderManager;
+import ekrut.server.managers.ServerReportManager;
+import ekrut.server.managers.ServerSessionManager;
+import ekrut.server.managers.ServerShipmentManager;
+import ekrut.server.managers.ServerTicketManager;
+import ocsf.server.AbstractServer;
+import ocsf.server.ConnectionToClient;
+
+public class EKrutServer extends AbstractServer{
+	
+	private DBController dbCon;
+	private String url, username, password;
+	private ServerSessionManager serverSessionManager;
+	private ServerTicketManager serverTicketManager;
+	private ServerOrderManager serverOrderManager;
+	private ServerInventoryManager serverInventoryManager;
+	private ServerReportManager serverReportManager;
+	private ServerShipmentManager serverShipmentManager;
+	
+	public EKrutServer(int port) {
+		super(port);
+		dbCon = new DBController(url, username, password);
+		dbCon.connect(); //need to check return value		
+		serverSessionManager = new ServerSessionManager(dbCon);
+		serverInventoryManager = new ServerInventoryManager(dbCon);
+		serverTicketManager = new ServerTicketManager(dbCon);
+		serverOrderManager = new ServerOrderManager(dbCon);
+		serverShipmentManager = new ServerShipmentManager(dbCon);
+	}
+
+	@Override
+	protected void handleMessageFromClient(Object msg, ConnectionToClient client) {
+		if(msg instanceof UserRequest) {
+			handleMessageUser((UserRequest)msg, client);
+		}
+		else if(msg instanceof TicketRequest) {
+			//Wait for the initialize of serverTicketManager class
+			//handleMessageTicket((TicketRequest)msg, client);
+		}
+		else if(msg instanceof InventoryItemRequest) {
+			handleMessageInventory((InventoryItemRequest)msg, client);
+		}
+		else if(msg instanceof OrderRequest) {
+			//Wait for the initialize of serverOrderManager class
+			//handleMessageOrder((OrderRequest)msg, client);
+		}
+		else if(msg instanceof ShipmentRequest) {
+			//handleMessageShipment((ShipmentRequest)msg, client);
+		}
+	}
+
+	private void handleMessageUser(UserRequest userRequest, ConnectionToClient client) {
+		UserResponse userResponse = null;
+		switch(userRequest.getAction()) {
+			case LOGIN:	
+				userResponse = serverSessionManager.loginUser(username,password,client);
+				break;
+			case LOGOUT:	
+				userResponse = serverSessionManager.logoutUser(username);
+				break;
+			case IS_LOGGEDIN:
+				String res = Boolean.toString(serverSessionManager.isLoggedin(username));
+				userResponse = new UserResponse(res);
+				break;
+			default:
+				break;
+		}
+		try {
+			client.sendToClient(userResponse);

```

Comment:
If none of the cases matched this would send null. You should instead return a result with `UNKNOWN_ERROR`. Same for the other methods.

Reply:
Your commit that sends UNKNOWN_ERROR in this situation is missing the actual code changes.